src/ol/reproj/Tile.js

```diff
@@ -103,7 +103,7 @@ class ReprojTile extends Tile {
 
     /**
      * @private
-     * @type {!Array<import("../Tile.js").default>}
+     * @type {!Array<ReprojTile|import("../ImageTile.js").default>}
```

Comment:
Use of `tile.getImage()` below doesn't work with the base tile class.

Reply:
Can source tiles be `ReprojTile`s?

I believe the return type of `getTileInternal` should be `TileImage` only.
https://github.com/openlayers/openlayers/blob/66fb11b31679e865711884c19fd48eeeee8b4ca7/src/ol/source/TileImage.js#L356-L365
`createTile_` always returns `TileImage` and the cache should also only contains this type.

Reply:
@MoonE - Thanks.  I think #14174 addresses your suggestion.

Reply:
Ooops.  I missed #14173.